File: 0796-rotate-string/0796-rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
        
        int length = s.length();
        s = s + s;

        for(int i = 0 ; i < length ; i++)
        {
            if(s.substring(i, i + length).equals(goal))
                return true;
        }

        return false;
    }
}

File: 0796-rotate-string/0796-rotate-string.js
/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
var rotateString = function(s, goal) {
    
    let left = 0;
    let right = s.length - 1;
    s = s + s;
    
    while(right < s.length)
    {
        let str = s.slice(left, right + 1);
        
        if(str === goal)
            return true;
        
        left++;
        right++;
    }
    
    return false;
};